libraries/AP_GPS/AP_GPS_NMEA.cpp

```diff
@@ -794,6 +873,150 @@ void AP_GPS_NMEA::parse_versiona_field(uint16_t term_number, const char *term)
 }
 #endif // AP_GPS_NMEA_UNICORE_ENABLED
 
+#if AP_GPS_NMEA_QUECTEL_ENABLED
+/*
+  Parse PQTM message fields for Quectel GNSS modules
+ */
+
+/*
+  parse PQTMVER field - Firmware Version Output
+  Example: $PQTMVER,1,MODULE,LG290P03AANR01A03S,2024/04/30,10:53:07*32
+ */
+void AP_GPS_NMEA::parse_pqtmver_field(uint16_t term_number, const char *term)
+{
+    auto &pv = _pqtmver;
+    
+    switch (term_number) {
+    case 1: // Version string (VerStr)
+        strncpy(pv.version, term, sizeof(pv.version)-1);
+        pv.version[sizeof(pv.version)-1] = 0;
```

Comment:
```suggestion
        strncpy(pv.version, term, sizeof(pv.version));
```

Our strncpy guarantees null termination.